app/services/checkout_service.rb

```diff
@@ -37,19 +41,45 @@ def shipping
     end
   end
 
-  private
-
   def cart_params
     @params.require(:cart).permit(:use_billing,
                                   billing: %i[first_name last_name address city zipcode country phone],
                                   shipping: %i[first_name last_name address city zipcode country phone])
   end
 
+  def credit_card_params
+    @params.require(:cart).require(:credit_card).permit(%i[number name_on_card month_year cvv])
+  end
+
   def address_params_type(type)
     type = cart_params[:use_billing] ? :billing : type
     cart_params.require(type)
   end
 
+  def update_delivery
+    @cart.update(delivery_id: @params[:cart][:delivery_id])
+  end
+
+  def delivery
+    Delivery.all
+  end
```

Comment:
Since this method returns collection, can we rename it to `deliveries`? Not a big deal, but a good practice

Reply:
Maybe it would be a good practice if it were not defined as WickedWizard step in controller. Thus, if we were to change ```ruby delivery ``` method we would have to change all the logic inside the checkout controller. Additionally, it'd be odd name for the Wizard step named in plural form.
Correct me if I'm wrong. I might have mistaken regarding this point.